Add OSD Node.js compatibility and instructions for using alternative version of Node.js runtime binary (#6232)
opensearch-project/documentation-website: _install-and-configure/install-dashboards/index.md

Other Chromium-based browsers might work, as well. Internet Explorer and Microsoft Edge Legacy are **not** supported.

## Node.js compatibility

> OpenSearch Dashboards requires the Node.js runtime binary to run. One is included in the distribution packages available from the [OpenSearch downloads page](https://opensearch.org/downloads.html){:target='\_blank'}.

Reviewer: ```suggestion
OpenSearch Dashboards requires the Node.js runtime binary to run. One is included in the distribution packages available from the [OpenSearch downloads page](https://opensearch.org/downloads.html){:target='\_blank'}.

```

> OpenSearch Dashboards requires the Node.js runtime binary to run and one is included in the distribution packages available from the [OpenSearch downloads page](https://opensearch.org/downloads.html){:target='\_blank'}.

Reviewer 2: ```suggestion
OpenSearch Dashboards requires the Node.js runtime binary to run. One is included in the distribution packages available from the [OpenSearch downloads page](https://opensearch.org/downloads.html){:target='\_blank'}.
```

> OpenSearch Dashboards 2.8.0 and newer can use Node.js versions 14, 16, and 18. The distribution packages for OpenSearch Dashboards 2.10.0 and newer include Node.js 18 and 14 (for backward compatibility). 

Reviewer 2: Insert a line space between lines 36 and 37.

> To use a Node.js runtime binary other that the ones included in the distribution packages, follow these steps:

Reviewer 2: ```suggestion
To use a Node.js runtime binary other than the ones included in the distribution packages, follow these steps:
```

1. Download and install [Node.js](https://nodejs.org/en/download){:target='\_blank'}; the compatible versions are `>=14.20.1 <19`.
2. Set the installation path to the `NODE_HOME` or `NODE_OSD_HOME` environment variables.
   

>    On UNIX, if Node.js is installed to `/usr/local/nodejs` and the runtime binary is `/usr/local/nodejs/bin/node`:

Reviewer 2: ```suggestion
    - On UNIX, if Node.js is installed to `/usr/local/nodejs` and the runtime binary is `/usr/local/nodejs/bin/node`:
```

    ```bash
    export NODE_HOME=/usr/local/nodejs
   ```

>    If Node.js installed using NVM and the runtime binary is `/Users/user/.nvm/versions/node/v18.19.0/bin/node`:

Reviewer 2: ```suggestion
    - If Node.js is installed using NVM and the runtime binary is `/Users/user/.nvm/versions/node/v18.19.0/bin/node`:
```

   ```bash
   export NODE_HOME=/Users/user/.nvm/versions/node/v18.19.0
   # or, if NODE_HOME is used for something else:
   export NODE_OSD_HOME=/Users/user/.nvm/versions/node/v18.19.0
   ```

>    On Windows, if Node.js is installed to `C:\Program Files\nodejs` and the runtime binary is `C:\Program Files\nodejs\node.exe`:

Reviewer 2: ```suggestion
    - On Windows, if Node.js is installed to `C:\Program Files\nodejs` and the runtime binary is `C:\Program Files\nodejs\node.exe`:
```

   ```powershell
   set "NODE_HOME=C:\Program Files\nodejs"
   # or using PowerShell:
   $Env:NODE_HOME = 'C:\Program Files\nodejs'
   ```

>    Please consult your operating system's documentation to make a persistent change to the environment variables.

Reviewer: ```suggestion
   Consult your operating system's documentation to make a persistent change to the environment variables.
```

>    Please consult your operating system's documentation to make a persistent change to the environment variables.

Reviewer 2: ```suggestion
View your operating system's documentation to make a persistent change to the environment variables.
{: .note}
```

> `bin/opensearch-dashboards`, the OpenSearch Dashboards start script will look for the Node.js runtime binary using `NODE_OSD_HOME`,

Reviewer: ```suggestion
The OpenSearch Dashboards start script,`bin/opensearch-dashboards`, searches for the Node.js runtime binary using `NODE_OSD_HOME`,
```

> `bin/opensearch-dashboards`, the OpenSearch Dashboards start script will look for the Node.js runtime binary using `NODE_OSD_HOME`,

Reviewer 2: ```suggestion
`bin/opensearch-dashboards`, the OpenSearch Dashboards start script, looks for the Node.js runtime binary using `NODE_OSD_HOME`,
```

Reviewer 2: Combine lines 65 and 66, as they are one sentence.

> and later `NODE_HOME`, before using the binaries included with the distribution packages. If a usable Node.js runtime binary is not found, the start script will

Reviewer 2: ```suggestion
and then `NODE_HOME`, before using the binaries included with the distribution packages. If a usable Node.js runtime binary is not found, the start script will
```

> attempt to find one in the system-wide `PATH` before failing.

Reviewer 2: Combine lines 65-67 into one line. There are breaks between words in sentences.

2. Set the installation path to the `NODE_HOME` or `NODE_OSD_HOME` environment variables.
   
    - On UNIX, if Node.js is installed to `/usr/local/nodejs` and the runtime binary is `/usr/local/nodejs/bin/node`:
    ```bash
    export NODE_HOME=/usr/local/nodejs
   ```

    - If Node.js is installed using NVM and the runtime binary is `/Users/user/.nvm/versions/node/v18.19.0/bin/node`:
   ```bash
   export NODE_HOME=/Users/user/.nvm/versions/node/v18.19.0
   # or, if NODE_HOME is used for something else:
   export NODE_OSD_HOME=/Users/user/.nvm/versions/node/v18.19.0
   ```

    - On Windows, if Node.js is installed to `C:\Program Files\nodejs` and the runtime binary is `C:\Program Files\nodejs\node.exe`:
   ```powershell
   set "NODE_HOME=C:\Program Files\nodejs"
   # or using PowerShell:
   $Env:NODE_HOME = 'C:\Program Files\nodejs'
   ```

   Consult your operating system's documentation to make a persistent change to the environment variables.

The OpenSearch Dashboards start script,`bin/opensearch-dashboards`, searches for the Node.js runtime binary using `NODE_OSD_HOME`,

> and then `NODE_HOME`, before using the binaries included with the distribution packages. If a usable Node.js runtime binary is not found, the start script will

Reviewer: ```suggestion
and then `NODE_HOME`, before using the binaries included with the distribution packages. If a usable Node.js runtime binary is not found, the start script will attempt to find one in the system-wide `PATH` before failing.
```

> attempt to find one in the system-wide `PATH` before failing.

Reviewer: ```suggestion
```
